servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/Publisher.java

```diff
@@ -118,6 +118,26 @@ protected Publisher() {
         return new MapPublisher<>(this, mapper);
     }
 
+    /**
+     * Cast this {@link Publisher} from type {@link T} to type {@link R}.
+     * <p>
+     * This method provides a data transformation in sequential programming similar to:
+     * <pre>{@code
+     *     List<R> results = ...;
+     *     for (T t : resultOfThisPublisher()) {
+     *         results.add(clazz.cast(t));
+     *     }
+     *     return results;
+     * }</pre>
+     * @param clazz The type to cast to.
+     * @param <R> The resulting type of the cast operation.
+     * @return The cast of this {@link Publisher} to type {@link R}.
+     * @see <a href="https://reactivex.io/documentation/operators/map.html">ReactiveX cast operator.</a>
+     */
+    public final <R> Publisher<R> cast(Class<R> clazz) {
```

Comment:
these operators are not impossible for users todo them selves but are commonly provided by other RS implementations and lowers the barrier to entry.